### ISSUE TITLE
Retry doesn't seem to work

### ISSUE THREAD

**Issue by PattaBov** [joshmossas/event-source-plus#12]
**Environment**
V.0.1.2.
Using it on React + Typescript

**Reproduction**

Here's my code to reproduce it:
```typescript
import { EventSourcePlus } from "event-source-plus";

const url = "YOUR_SSE_ENDPOINT";

const eventSource = new EventSourcePlus(url, {
    headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer YOUR_TOKEN`
    },
    maxRetryCount: 10,
    maxRetryInterval: 3000,
    keepalive: true
});

const controller = eventSource.listen({
    onResponse(res) {
        if (res.error || !res.response?.ok) {
            console.error("[ON_RESPONSE_ERROR]", res.error || `Status: ${res.response?.status}`);
        } else {
            console.info("Subscribed to Server-sent Event with status code " + res.response.status);
        }
    },
    onMessage(event) {
        console.info("[MESSAGE]", event.data);
    },
    onRequest({ request, options }) {
        console.info("[ON_REQUEST]", request, options);
    },
    onRequestError({ error }) {
        console.error("[ON_REQUEST_ERROR]", error);
    },
    onResponseError({ response }) {
        console.error("[ON_RESPONSE_ERROR]", `Status: ${response.status}`, response);
    },
    onClose() {
        console.info("[CONNECTION_CLOSED]");
    }
});
``` 

Steps to reproduce: 

1. Set up an SSE endpoint.
2. Connect to the SSE endpoint using `EventSourcePlus` with the configuration above.
3. Observe the console logs for repeated connection attemps even after a successful connection.

**Describe the bug**

The retry mechanism in the event-source-plus package appears to enter an infinite loop, repeatedly reconnecting even after a successful connection. This is due to the _handleRetry method calling _handleConnection, which resets the retry count and continues to retry without checking if the connection was successful.

**Additional context**

Maybe I miss something in the configuration and in that case I'm sorry about the bug creation. Server sent events is really interesting for our project but I cannot make it work properly.

**Comment by joshmossas** [comment]
Hey @PattaBov could you provide some context as to what your server code looks like? Unfortunately I cannot reproduce with the client code alone. I need to know what the server is sending to debug.

Or at the very least can you show what the headers and body returned by the server in `onResponse` are?

This library will auto retry if any of the following conditions are met:
- The server returns a status code that is not between `200-299`
- The server doesn't return the `"Content-Type: text/event-stream"` header

**Comment by joshmossas** [comment]
Also for context the hooks trigger in this order:

```bash
onRequest
onRequestError # if there's an error constructing the request
onResponse
onResponseError # if there's an error in the response from the server (i.e. malformed request, bad http status code, missing header(s))
```

**Comment by PattaBov** [comment]
Hello and thank you for your answer!

Indeed my server doesn't send `"Content-Type: text/event-stream"` in the response. But I managed to see that it still works with `maxRetryCount` being set to 1.

I think the problem here is that when `_handleRetry` is called, then it creates a new AbortController and calls `_handleConnection` again. In this method on line 142, the `retryCount` is reset to 0. So even if I face multiple failing responses, the `maxRetryCount` I first set is never reached.

From what I understand, if my response doesn't include `"Content-Type: text/event-stream"`, then it throws an error and indeed in the catch block it triggers `_handleRetry`.

**Comment by joshmossas** [comment]
Okay yeah line 141 is supposed to throw an error if any of the error conditions are met. So normally, line 142 is never reached if any kind of error is returned by the server.

But wow you're right if the `Content-Type` header is `undefined` it will not throw an error causing a potential infinite loop. This is definitely an oversight. Let me put together a patch really quick and get an update out. I'll add another integration test to cover this scenario as well to prevent potential regressions in the future.

**Comment by joshmossas** [comment]
@PattaBov the fix for this is live in `v0.1.3` please upgrade to that and let me know if you are still having issues.
